app/src/main/java/org/schabi/newpipe/player/BasePlayer.java

```diff
@@ -1285,4 +1285,22 @@ private boolean isPlaybackResumeEnabled() {
         return prefs.getBoolean(context.getString(R.string.enable_watch_history_key), true)
                 && prefs.getBoolean(context.getString(R.string.enable_playback_resume_key), true);
     }
+
+    public void setTimer(int hourOfDay, int minute) {
+        Toast.makeText(this.context, this.context.getString(R.string.invalid_method_call) + "setTimer()", Toast.LENGTH_SHORT).show();
+    }
+
+    public void cancelTimer() {
+        Toast.makeText(this.context, this.context.getString(R.string.invalid_method_call) + "cancelTimer()", Toast.LENGTH_SHORT).show();
+    }
+
+    public int getHourOfDay() {
+        Toast.makeText(this.context, this.context.getString(R.string.invalid_method_call) + "getHourOfDay()", Toast.LENGTH_SHORT).show();
+        return 0;
+    }
+
+    public int getMinutes() {
+        Toast.makeText(this.context, this.context.getString(R.string.invalid_method_call) + "getMinutes()", Toast.LENGTH_SHORT).show();
+        return 0;
+    }
```

Comment:
I am unsure about this. Showing a toast might confuse the user. @Stypox @avently can you please take a look at it? Do you have suggestions for this?

Reply:
@javed-hussain
An `UnsupportedOperationException` should be thrown, or maybe you should remove altogether the base functions in the BasePlayer class, since they are specific for the background player and nothing else. Also, note that there is a missing `@Override` before the function.

Reply:
Another option would be to make the Timer available in every player. T

Reply:
Yes that's why I added the functions in base class and thrown the exception as Currently other players are not supported and overridden the same in `BackgroundPlayer` class only.